### PR TITLE
Add references to the Cloud costs page from cdc docs

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-10414

This PR:

- Updates the Understand Cloud Costs page to include a short definition of "watched" as per the cdc docs.
- Updates the feature capitalization on the same page.
- Adds links back to the Understand Cloud Costs page from the Create and Configure Changefeeds page.
- Adds a note to the Changefeed Best Practices page to remind users to consider Cloud Costs when planning changefeeds.

### CODE REVIEW

**Review by rohan-joshi — APPROVED** [review]
lgtm

File: src/current/cockroachcloud/costs.md [hunk 1]
```diff
@@ -289,12 +289,16 @@ In CockroachDB {{ site.data.products.basic }}, CDC cost is usage-based via Reque
...
+
+In CockroachDB {{ site.data.products.standard }}, CDC is billed monthly based on usage, determined by the total GiB-Month of watched across all of a cluster’s changefeeds. The per-GiB unit price is tiered, based on the total watched: Less than 5 GiB-Month, 5 to 100 GiB-Month, 100 to 250 GiB-Month, 250 to 500 GiB-Month, or 500 GiB-Month and higher.
 
 </section>
 
 <section class="filter-content" markdown="1" data-scope="advanced">
 
+{% include common/define-watched-cdc.md %}
+
 In CockroachDB {{ site.data.products.advanced }}, CDC is billed monthly based on usage, determined by the total GiB-Month watched across all of a cluster’s changefeeds and whether the Advanced security add-on is enabled. The per-GiB unit price is tiered, based on the total watched: Less than 5 GiB-Month, 5 to 100 GiB-Month, 100 to 250 GiB-Month, 250 to 500 GiB-Month, or 500 GiB-Month and higher.
```
Comment (mikeCRL) [inline src/current/cockroachcloud/costs.md:302]:
```suggestion
In CockroachDB {{ site.data.products.advanced }}, CDC is billed monthly based on the total size of a cluster's watched tables and whether the {{ site.data.products.advanced }} security add-on is enabled. The per-GiB unit price is tiered, based on the size of watched data: Less than 5 GiB-Month, 5 to 100 GiB-Month, 100 to 250 GiB-Month, 250 to 500 GiB-Month, or 500 GiB-Month and higher.
```

**Review by mikeCRL — APPROVED** [review]
Looks good. Some suggestions for your consideration.

**Comment by kathancox** [comment]
Nice thanks @mikeCRL, added your suggestions!

**Pull request merged**